### PR TITLE
Introduce common interface to start the domains

### PR DESCRIPTION
**Pull request by oleksiimoisieiev**
Common interface allows users to start the domains providing domain_config which includes all needed parameters as well as the image and dtb buffers.
Also added calculation of the appropriate dtb address in the Domain ram

### CODE REVIEW

File: xen-dom-mgmt/include/xen_dom_mgmt.h [hunk 1]
```diff
@@ -16,4 +17,9 @@ int domu_destroy(const struct shell *shell, size_t argc, char **argv);
 int domu_pause(const struct shell *shell, size_t argc, char **argv);
 int domu_unpause(const struct shell *shell, size_t argc, char **argv);
 
+int domain_create(const struct shell *shell, struct xen_domain_cfg *domcfg,
```
Comment (firscity) [inline xen-dom-mgmt/include/xen_dom_mgmt.h:20]: Actually shell parameter for functions below is redundant and should be removed. This only required to implement callback for Zephyr shell cmds.

File: xen-dom-mgmt/src/xen-dom-mgmt.c [hunk 2]
```diff
@@ -595,29 +586,12 @@ int domu_create(const struct shell *shell, size_t argc, char **argv)
...
-		return -EINVAL;
-	}
-
-	domid = parse_domid(argc, argv);
-	if (!domid) {
-		printk("Invalid domid passed to create cmd\n");
-		return -EINVAL;
-	}
-
 	/*domcfg = (domid == DOMID_DOMD) ? &domd_cfg : &domu_cfg;
```
Comment (firscity) [inline xen-dom-mgmt/src/xen-dom-mgmt.c:667]: This should be removed too.

File: xen-dom-mgmt/src/xen-dom-mgmt.c [hunk 3]
```diff
@@ -757,6 +726,28 @@ int domu_create(const struct shell *shell, size_t argc, char **argv)
 	return rc;
 }
 
+int domu_create(const struct shell *shell, int argc, char **argv)
+{
+	uint32_t domid;
+
+	if (argc < 3 || argc > 4) {
```
Comment (firscity) [inline xen-dom-mgmt/src/xen-dom-mgmt.c:809]: `argc != 3`?

File: xen-dom-mgmt/src/xen-dom-mgmt.c [hunk 4]
```diff
@@ -813,11 +793,9 @@ int domu_destroy(const struct shell *shell, size_t argc, char **argv)
...
 }
 
-int domu_pause(const struct shell *shell, size_t argc, char **argv)
+int domu_destroy(const struct shell *shell, size_t argc, char **argv)
 {
-	int rc;
 	uint32_t domid = 0;
-	struct xen_domain *domain = NULL;
 
 	if (argc < 3 || argc > 4) {
```
Comment (firscity) [inline xen-dom-mgmt/src/xen-dom-mgmt.c:876]: `argc != 3`

File: xen-dom-mgmt/src/xen-dom-mgmt.c [hunk 5]
```diff
@@ -757,6 +726,28 @@ int domu_create(const struct shell *shell, size_t argc, char **argv)
...
+		return -EINVAL;
+	}
+
+	domid = parse_domid(argc, argv);
+	if (!domid) {
+		printk("Invalid domid passed to create cmd\n");
+		return -EINVAL;
+	}
+
+	return domain_create(shell, &domd_cfg, domid);
```
Comment (firscity) [inline xen-dom-mgmt/src/xen-dom-mgmt.c:824]: `&domd_cfg` should be replaced with more generic way, please add TODO here
Not all domains should use this config.

**Review by firscity — CHANGES_REQUESTED** [review]
Also, please make commit message more coherent and consistent.

File: xen-dom-mgmt/src/xen-dom-mgmt.c (hunk 5 above)
Comment (oleksiimoisieiev) [reply xen-dom-mgmt/src/xen-dom-mgmt.c]: +

File: xen-dom-mgmt/src/xen-dom-mgmt.c (hunk 3 above)
Comment (oleksiimoisieiev) [reply xen-dom-mgmt/src/xen-dom-mgmt.c]: +

File: xen-dom-mgmt/src/xen-dom-mgmt.c (hunk 4 above)
Comment (oleksiimoisieiev) [reply xen-dom-mgmt/src/xen-dom-mgmt.c]: +

File: xen-dom-mgmt/src/xen-dom-mgmt.c (hunk 2 above)
Comment (oleksiimoisieiev) [reply xen-dom-mgmt/src/xen-dom-mgmt.c]: +

File: xen-dom-mgmt/include/xen_dom_mgmt.h (hunk 1 above)
Comment (oleksiimoisieiev) [reply xen-dom-mgmt/include/xen_dom_mgmt.h]: +

**Review by firscity — APPROVED** [review]
`Reviewed-by: Dmytro Firsov <dmytro_firsov@epam.com>`

**Pull request merged**